index.js

```diff
@@ -61,181 +65,226 @@ DESCRIPTION
 }
 
-// filters for getting rules and issues
-let filterRule = DEFAULT_FILTER;
-let filterIssue = DEFAULT_FILTER;
+(async () => {
+  var severity = new Map();
+  severity.set('MINOR', 0);
+  severity.set('MAJOR', 1);
+  severity.set('CRITICAL', 2);
+  severity.set('BLOCKER', 3);
+
+  const data = {
+    date: new Date().toDateString(),
+    projectName: argv.project,
+    applicationName: argv.application,
+    releaseName: argv.release,
+    branch: argv.branch,
+    sinceLeakPeriod: (argv.sinceleakperiod == 'true'),
+    previousPeriod: '',
+    allBugs: (argv.allbugs == 'true'),
+    fixMissingRule: (argv.fixMissingRule == 'true'),
+    noSecurityHotspot: (argv.noSecurityHotspot == 'true'),
+    // sonar URL without trailing /
+    sonarBaseURL: argv.sonarurl.replace(/\/$/, ""),
+    sonarOrganization: argv.sonarorganization,
+    rules: [],
+    issues: []
+  };
 
-if(data.allBugs){
-  filterRule = "";
-  filterIssue = "";
-}
+  const leakPeriodFilter = data.sinceLeakPeriod ? '&sinceLeakPeriod=true' : '';
+  data.deltaAnalysis = data.sinceLeakPeriod ? 'Yes' : 'No';
+  const sonarBaseURL = data.sonarBaseURL;
+  const sonarComponent = argv.sonarcomponent;
+  const withOrganization = data.sonarOrganization ? `&organization=${data.sonarOrganization}` : '';
+  var headers = {};
+
+  let DEFAULT_FILTER="";
+  let OPEN_STATUSES="";
+  // Default filter gets only vulnerabilities
+  if(data.noSecurityHotspot){
+    // For old versions of sonarQube (sonarQube won't accept filtering on a type that doesn't exist and will give HTTP 400 {"errors":[{"msg":"Value of parameter 'types' (SECURITY_HOTSPOT) must be one of: [CODE_SMELL, BUG, VULNERABILITY]"}]})
+    DEFAULT_FILTER="&types=VULNERABILITY"
+    OPEN_STATUSES="OPEN,CONFIRMED,REOPENED"
+  }
+  else{
+    // For newer versions of sonar, rules and issues may be of type VULNERABILITY or SECURITY_HOTSPOT
+    DEFAULT_FILTER="&types=VULNERABILITY,SECURITY_HOTSPOT"
+    // the security hotspot adds TO_REVIEW,IN_REVIEW
+    OPEN_STATUSES="OPEN,CONFIRMED,REOPENED,TO_REVIEW,IN_REVIEW"
+  }
 
-if(data.branch){
-  filterIssue=filterIssue + "&branch=" + data.branch
-}
+  // filters for getting rules and issues
+  let filterRule = DEFAULT_FILTER;
+  let filterIssue = DEFAULT_FILTER;
 
-if(data.fixMissingRule){
-  filterRule = "";
-}
+  if(data.allBugs){
+    filterRule = "";
+    filterIssue = "";
+  }
+
+  if(data.branch){
+    filterIssue=filterIssue + "&branch=" + data.branch
+  }
+
+  if(data.fixMissingRule){
+    filterRule = "";
+  }
+
+  var proxy = null;
+  // the tunnel agent if a forward proxy is required, or remains null
+  var agent = null;
+  // Preparing configuration if behind proxy
+  if (process.env.http_proxy){
+    proxy = process.env.http_proxy;
+    var url = new URL(proxy);
+    var proxyHost = url.hostname;
+    var proxyPort = url.port;
+    console.error('using proxy %s:%s', proxyHost, proxyPort);
```

Comment:
Perhaps use info instead
```suggestion
    console.info('using proxy %s:%s', proxyHost, proxyPort);
```

Reply:
@bboutier thanks for your feedback. I've found that error output gets printed in the standard output of the node command, while info gets logged into the html report; it's why i've used error level